src/Http/Middleware/DoctrineMiddleware.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Http\Middleware;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class DoctrineMiddleware implements MiddlewareInterface
+{
+    /**
+     * @var ManagerRegistry
+     */
+    private $managerRegistry;
+
+    public function __construct(ManagerRegistry $managerRegistry)
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        foreach ($this->managerRegistry->getManagers() as $name => $manager) {
+            assert($manager instanceof EntityManagerInterface);
+
+            /** @var \Doctrine\DBAL\Connection $connection */
+            $connection = $manager->getConnection();
```

Comment:
I think doing this will force the DIC to always create connection instances but maybe the current worker will not use the database during it's whole life.

Connections services could be retrieved with `ManagerRegistry::getConnectionNames()`, and you need to inject the Symfony container to check if the service is initialized before retrieving it, like here: https://github.com/doctrine/DoctrineBundle/blob/master/Registry.php#L62

Reply:
I checked it before PR, and re-check it localy (with PDO driver) now, `getConnection` not create connection to database, until `Doctrine\DBAL\Connection::connect` not called

Reply:
You are right, the connection is not opened on the database, but iterating on managers, et getting connections forces the Symfony DIC to resolve the doctrine stack on the very first request.

I would prefer a solution where we first check if the service has been initialized by the container before accessing it. For example like it's done here: https://github.com/doctrine/DoctrineBundle/blob/master/DoctrineBundle.php#L125

We can rely on `doctrine.connections` parameters or use `ManagerRegistry::getConnectionNames()` (which are the same).

Not relying on `getManagers` has also another effect: people using DoctrineBundle only for DBAL (no ORM), will also enjoy this middleware.

I'm sorry for beeing annoying on this, but I try to keep workers initialization as cheap as possible.

Reply:
You absolutely right, fixed.